### PR TITLE
Nemaniarjun/flush+refresh

### PR DESCRIPTION
**Pull request by nemaniarjun**

### CODE REVIEW

**Comment by abhsag24** [comment]
Hi thank you for your Patch make sure you have the right commit structure that is 
```
tag: Shortlog

body
Fixes <issue number/url>
```

File: uiplib/gui/mainGui.py [hunk 1]
```diff
@@ -87,6 +88,7 @@ def set_wallpaper(self):
     def flush(self):
         """Method to flush all images."""
         print("Flush Clicked!")
```
Comment (abhsag24) [inline uiplib/gui/mainGui.py]: remove this print?

File: uiplib/utils/utils.py [hunk 2]
```diff
@@ -42,3 +44,13 @@ def check_sites(settings):
                 else:
                     sites_present[key] = True
     return sites_present
+
+
+def flush_wallpapers(settings):
+    """Delete all downloaded wallpapers."""
+    print("Deleting all downloaded wallpapers...")
+    try:
+        shutil.rmtree(settings['pics-folder'])
```
Comment (abhsag24) [inline uiplib/utils/utils.py]: so i know this is how it was done earlier, but consider this scenario: 
A user sets the pics folder to one of his personal directories
you delete everything on that directory :P

Comment (abhsag24) [inline uiplib/utils/utils.py]: i know this is how it was done earlier, but can we have something else? atleast a warning?

File: uiplib/gui/mainGui.py (hunk 1 above)
Comment (abhsag24) [inline uiplib/gui/mainGui.py]: warning goes here if ok is clicked

File: uiplib/gui/mainGui.py [hunk 3]
```diff
@@ -92,6 +92,7 @@ def flush(self):
 
     def update_images(self):
         """Method to get images from directory."""
+        print("Updating Image List")
```
Comment (abhsag24) [inline uiplib/gui/mainGui.py]: why isnt flush called? what do you propose the usecase of the refresh button? just refresh of the UI, shouldnt this happen when flush is clicked itself?

File: uiplib/gui/mainGui.py (hunk 1 above)
Comment (nemaniarjun) [inline uiplib/gui/mainGui.py]: So a warning via GUI? okay.

File: tests/test_utils.py [hunk 4]
```diff
@@ -9,8 +9,24 @@
 
 class UtilsTest(unittest.TestCase):
 
+    def test_flush_wallpapers(self):
+        testdir = os.path.join(os.path.expanduser("~"), '.WAKEUPDOLORES')
+        setupUtils.make_dir(testdir)
+        test_dir_to_delete = os.path.join(testdir, 'tobedeleted')
```
Comment (abhsag24) [inline tests/test_utils.py]: why create 2 dirs?

File: tests/test_utils.py [hunk 5]
```diff
@@ -9,8 +9,24 @@
...
+        setupUtils.make_dir(test_dir_to_delete)
+        test_settings = {"pics-folder": testdir}
```
Comment (abhsag24) [inline tests/test_utils.py]: while here you only give the base dir

File: uiplib/UIP.py [hunk 6]
```diff
@@ -46,12 +45,7 @@ def main():
     if settings['offline']:
         print("You have choosen to run UIP in offline mode.")
     if settings['flush']:
-        print("Deleting all downloaded wallpapers...")
-        try:
-            shutil.rmtree(settings['pics-folder'])
-            make_dir(settings['pics-folder'])
-        except FileNotFoundError:
-            pass
+        flush_wallpapers(settings)
```
Comment (abhsag24) [inline uiplib/UIP.py]: i guess you should just pass pics-folder, no point in passingall of the settings

Comment (nemaniarjun) [inline uiplib/UIP.py]: Will do this.

File: tests/test_utils.py (hunk 4 above)
Comment (nemaniarjun) [inline tests/test_utils.py]: Creating a directory. Emptying it. Then checking if it is empty.

Comment (abhsag24) [inline tests/test_utils.py]: ahh ok 👍

**Pull request merged**